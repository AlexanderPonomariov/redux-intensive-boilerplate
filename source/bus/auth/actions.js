// Types
import { types } from './types';

export const authActions = Object.freeze({
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
});
