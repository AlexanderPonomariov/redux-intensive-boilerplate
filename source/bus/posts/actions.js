// Instruments
import { types } from './types';

export const postsActions = Object.freeze({
    fetchPosts: () => ({
        type: types.FETCH_POSTS,
    }),
    fetchPostsSuccess: (posts) => ({
        type:    types.FETCH_POSTS_SUCCESS,
        payload: posts,
    }),
    fetchPostsFail: (error) => ({
        type:    types.FETCH_POSTS_FAIL,
        payload: error,
        error:   true,
    }),

    createPost: (post) => ({
        type:    types.CREATE_POST,
        payload: post,
    }),
    removePost: (id) => ({
        type:    types.REMOVE_POST,
        payload: id,
    }),
    likePost: (likedPostIds) => ({
        type:    types.LIKE_POST,
        payload: likedPostIds,
    }),
    unlikePost: (unlikedPostIds) => ({
        type:    types.UNLIKE_POST,
        payload: unlikedPostIds,
    }),
});
