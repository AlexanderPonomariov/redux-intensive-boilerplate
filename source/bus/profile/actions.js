// Types
import { types } from './types';

export const profileActions = Object.freeze({
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user,
    }),
    updateAvatar: (avatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
});
