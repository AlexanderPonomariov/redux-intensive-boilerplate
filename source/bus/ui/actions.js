// Instruments
import { types } from './types';

export const uiActions = Object.freeze({
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    initializeSuccess: () => ({
        type: types.INITIALIZE_SUCCESS,
    }),
    setPostsFetchingState: (postsFetchingState) => ({
        type:    types.SET_POSTS_FETCHING_STATE,
        payload: postsFetchingState,
    }),
    setAuthFetchingState: (authFetchingState) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: authFetchingState,
    }),
    setOnlineState: (onlineState) => ({
        type:    types.SET_ONLINE_STATE,
        payload: onlineState,
    }),
    setProfileFetchingState: (state) => ({
        type:    types.SET_PROFILE_FETCHING_STATE,
        payload: state,
    }),
    setProfileEditingState: (state) => ({
        type:    types.SET_PROFILE_EDITING_STATE,
        payload: state,
    }),
    setAvatarFetchingState: (state) => ({
        type:    types.SET_AVATAR_EDITING_STATE,
        payload: state,
    }),
    setPasswordEditingState: (state) => ({
        type:    types.SET_PASSWORD_EDITING_STATE,
        payload: state,
    }),
});
