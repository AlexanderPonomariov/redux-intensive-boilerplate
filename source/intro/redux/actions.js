// Instruments
import { CHANGE_PHOTO } from './types';

export const changePhoto = (photo) => ({
    type:    CHANGE_PHOTO,
    payload: photo,
});
