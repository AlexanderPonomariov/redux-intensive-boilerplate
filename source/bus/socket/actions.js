// Instruments
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postsActions } from '../posts/actions';

export const socketActions = Object.freeze({
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState(true));
        });

        socket.on('disconnect', () => {
            dispatch(uiActions.setOnlineState(false));
        });
    },

    listenPosts: () => (dispatch, getState) => {
        socket.on('create', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.createPost(data));
        });

        socket.on('remove', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.removePost(data));
        });

        socket.on('like', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === data.userID)
                    .delete('avatar');

                dispatch(
                    postsActions.likePost({
                        postID: data.postID,
                        liker,
                    })
                );

                return;
            }

            dispatch(postsActions.unlikePost(data));
        });
    },
});
