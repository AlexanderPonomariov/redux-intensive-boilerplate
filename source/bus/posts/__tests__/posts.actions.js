// Instruments
import { postsActions } from '../actions';

describe('posts actions:', () => {
    test('FETCH_POSTS:', () => {
        expect(postsActions.fetchPosts()).toMatchSnapshot();
    });
});
