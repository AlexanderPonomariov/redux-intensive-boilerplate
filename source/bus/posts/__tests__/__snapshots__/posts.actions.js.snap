// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts actions: FETCH_POSTS: 1`] = `
Object {
  "type": "FETCH_POSTS",
}
`;
